### PR TITLE
Feat/kraken market orders

### PR DESCRIPTION
**Pull request by fengtality**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

- Add support for market orders to the Kraken connector
- Adds a sample script that places an order that can be used to test order placement functionality on various connectors.

### CODE REVIEW

File: hummingbot/connector/exchange/kraken/kraken_exchange.pyx [hunk 1]
```diff
@@ -865,7 +865,12 @@ cdef class KrakenExchange(ExchangeBase):
 
         except Exception as e:
             self.c_stop_tracking_order(order_id)
-            order_type_str = 'LIMIT' if order_type is OrderType.LIMIT else "LIMIT_MAKER"
+            if order_type is OrderType.LIMIT:
+                order_type_str = 'LIMIT'
+            elif order_type is OrderType.LIMIT_MAKER:
+                order_type_str = 'LIMIT_MAKER'
+            else:
+                order_type_str = 'MAKER'
```
Comment (cardosofede) [inline hummingbot/connector/exchange/kraken/kraken_exchange.pyx:873]: Are you sure that Market is MAKER in kraken?

File: hummingbot/connector/exchange/kraken/kraken_exchange.pyx [hunk 2]
```diff
@@ -822,7 +822,7 @@ cdef class KrakenExchange(ExchangeBase):
         try:
             order_result = None
             order_decimal_amount = f"{decimal_amount:f}"
-            if order_type is OrderType.LIMIT or order_type is OrderType.LIMIT_MAKER:
+            if order_type is OrderType.LIMIT or order_type is OrderType.LIMIT_MAKER or order_type is OrderType.MARKET:
```
Comment (cardosofede) [inline hummingbot/connector/exchange/kraken/kraken_exchange.pyx:825]: here you can simplify (mainly if other types are added) with
if order_type in self.supported_order_types() that you added in line 777

**Review by cardosofede — CHANGES_REQUESTED** [review]
I think that also you need to adapt the test

File: hummingbot/connector/exchange/kraken/kraken_exchange.pyx (hunk 1 above)
Comment (fengtality) [reply hummingbot/connector/exchange/kraken/kraken_exchange.pyx:873]: good catch this should be `MARKET`. It's just used in the log message below.

**Comment by rapcmia** [comment]
Run the market order script [here](https://gist.github.com/rapcmia/9c81c1da68d60be7e6a57d717d318f4f)
- Getting `OSError: {'error': {'error': ['EOrder:Invalid price:Invalid price argument.']}}`
  ```
  2023-12-08 12:00:32,224 - 4703 - hummingbot.connector.exchange.kraken.kraken_exchange - ERROR - Error received from https://api.kraken.com/0/private/AddOrder. Response is {'error': ['EOrder:Invalid price:Invalid price argument.']}.
  2023-12-08 12:00:32,225 - 4703 - hummingbot.connector.exchange.kraken.kraken_exchange - NETWORK - Error submitting buy MAKER order to Kraken for 0.20000000 SOL-USDT NaN.
  Traceback (most recent call last):
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 837, in hummingbot.connector.exchange.kraken.kraken_exchange.KrakenExchange.execute_buy
      order_result = await self.place_order(userref=userref,
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 799, in place_order
      return await self._api_request_with_retry("post",
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 711, in _api_request_with_retry
      raise e
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 694, in hummingbot.connector.exchange.kraken.kraken_exchange.KrakenExchange._api_request_with_retry
      result= await self._api_request(method, endpoint, params, data, is_auth_required)
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 722, in _api_request
      async with self._throttler.execute_task(endpoint):
    File "hummingbot/connector/exchange/kraken/kraken_exchange.pyx", line 766, in hummingbot.connector.exchange.kraken.kraken_exchange.KrakenExchange._api_request
      raise IOError({"error": response_json})
  OSError: {'error': {'error': ['EOrder:Invalid price:Invalid price argument.']}}
  ``` 
   - Run the same script on other connectors that support market order, all ok but not with this PR

[logs_a.log](https://github.com/hummingbot/hummingbot/files/13608720/logs_a.log)

**Comment by fengtality** [comment]
@rapcmia I updated the Kraken connector - can you test it out, especially on V2 strategies (ie. Binanc/Kucoin candles + trading on Kraken)

**Review by nikspz — APPROVED** [review]
Test performed:
- Cloned and installed feature branch
- Ran Dman-v3 strategy using kraken and binance_perpetual as candles exchange
- Review bot failed to close orders with market orders (discussed to update later)
- Ran simple_order_example script on sell and buy, review market order was filled 
- Created/start pureMM using kraken, review order fills not affected

**Pull request merged**